apps/els_lsp/src/els_indexing.erl

```diff
@@ -50,9 +50,33 @@ find_and_index_file(FileName) ->
 
 -spec index_file(binary()) -> {ok, uri()}.
 index_file(Path) ->
-  try_index_file(Path, 'deep'),
+  SkipGeneratedFiles = els_config_indexing:get_skip_generated_files(),
+  GeneratedFilesTag = els_config_indexing:get_generated_files_tag(),
+  try_index_file(Path, 'deep', SkipGeneratedFiles, GeneratedFilesTag),
```

Comment:
The two values are always passed in together - should they be made into just one accessor? Something like:
```
els_config_indexing:skip_generated_config()
```
which would return `disabled | {enabled, Tag}`

It should simplify the code and follow the "make impossible states unrepresentable" approach

Reply:
I like this a lot, but let's do this as a follow up. The main reason is that the whole config management needs refactoring, so it would be nice to come up with a common pattern.